src/kv/kv.hpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * Canary - A free and open-source MMORPG server emulator
+ * Copyright (©) 2019-2022 OpenTibiaBR <opentibiabr@outlook.com>
+ * Repository: https://github.com/opentibiabr/canary
+ * License: https://github.com/opentibiabr/canary/blob/main/LICENSE
+ * Contributors: https://github.com/opentibiabr/canary/graphs/contributors
+ * Website: https://docs.opentibiabr.com/
+ */
+
+#pragma once
+
+#include <string>
+#include <mutex>
+#include <initializer_list>
+#include <parallel_hashmap/phmap.h>
+#include <optional>
+
+#include "lib/logging/logger.hpp"
+#include "value_wrapper.hpp"
+
+class KVBackend {
+public:
+	virtual std::optional<ValueWrapper> load(const std::string &key) = 0;
+	virtual bool save(const std::string &key, const ValueWrapper &value) = 0;
+	virtual ~KVBackend() = default;
+};
+
+class KVStore {
```

Comment:
I'm a bit confused by this class, I think we could simplify it:

The concepts of KVStore and KVBackend seems mixed to me, in the end they are both partially representing a repository/store. I think we can unify and simplify them, by presenting a more clear and simple interface/abstraction. What do you think?

Ideally I'd have:
- 1 backend/store/repository class that represents the interface which the user will use to iteract with KV.
- 1 implementation which abstract dependencies (mysql/pubsub/etc).

With that, ScopedKV could also be an implementation of the interface, that depends on another implementation (composition) and a scope.

Reply:
I think I addressed this satisfactorily in the last commit now :)